src/Components/ArtworkFilter/ArtworkFilters/ArtworkLocationFilter.tsx

```diff
@@ -20,3 +38,40 @@ export const ArtworkLocationFilter: React.FC<
     />
   )
 }
+
+export const EnhancedArtworkLocationFilter: React.FC<
+  React.PropsWithChildren<ArtworkLocationFilterProps>
+> = props => {
+  const { expanded } = props
+  const tracking = useTracking()
+  const { contextPageOwnerId, contextPageOwnerSlug, contextPageOwnerType } =
+    useAnalyticsContext()
+
+  const handleSelectAll = (state: FilterSelectChangeState) => {
+    const event: CommercialFilterSelectedAll = {
+      action: ActionType.commercialFilterSelectedAll,
+      context_module: ContextModule.artworkGrid,
+      context_owner_type: contextPageOwnerType,
+      context_owner_id: contextPageOwnerId,
+      context_owner_slug: contextPageOwnerSlug,
+      facet: "locationCities",
+      query: state.query,
+    }
+
+    tracking.trackEvent(event)
```

Comment:
As seen in the Segment debugger:

![Screenshot 2025-01-31 at 3 43 42 PM](https://github.com/user-attachments/assets/37f864a4-bd69-4d6b-97de-132d942a4a9c)